app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb

```diff
@@ -7,36 +7,24 @@ def self.miq_template_type
       parent::Parser::ParserDictionaryConstants::MIQ_TYPES["template"]
     end
 
-    def initialize(ems, _options = nil)
-      ems_auth = ems.authentications.first
-
-      @ems        = ems
-      @connection = ems.connect(:user => ems_auth.userid,
-                                :pass => ems_auth.password,
-                                :host => ems.endpoints.first.hostname,
-                                :port => ems.endpoints.first.port)
-      @parser     = init_parser(@connection)
+    def initialize(connection, _options = nil)
```

Comment:
:-1: for changing the initializer, this is a common pattern across all refresh_parsers.